src/databricks/labs/ucx/azure/access.py

```diff
@@ -512,9 +512,20 @@ def create_access_connectors_for_storage_accounts(self) -> list[tuple[AccessConn
         thread_name = "Creating access connectors for storage accounts"
         results, errors = Threads.gather(thread_name, tasks)
         if len(errors) > 0:
+            logger.error("Error creating all access connectors. Deleting incomplete ones.")
+            delete_access_connectors = [access_connector for access_connector, _ in results]
+            self.delete_access_connectors(delete_access_connectors)
             raise ManyError(errors)
         return list(results)
 
+    def delete_storage_credentials(self, storage_credentials: list[Any]):
```

Comment:
Depending on your preference:

```suggestion
    def delete_storage_credential(self, *storage_credentials: Any):
```

Same below